### ISSUE TITLE
_PENDING action not coming up when redux-logger is present

### ISSUE THREAD

**Issue by nicnilov** [pburtchaell/redux-promise-middleware#161]
I'm trying to use redux-promise-middleware v4.2.1 in my project and encountering this weird issue. This piece of code

    const logIn = (email, password) => ({
      type: types.LOG_IN,
      payload: api.login(email, password)
    })

where `api.login` returns a promise, gets called in this fashion from a React component:

    store.dispatch(logIn(this.state.email, this.state.password))

The promise gets resolved properly and the flow itself works, albeit with a kink. Although the `LOG_IN_FULFILLED` and `LOG_IN_REJECTED` actions do come through, `LOG_IN_PENDING` is not there, instead I am getting a plain `LOG_IN` one.

Here is a piece of the console log:

    action LOG_IN @ 01:53:31.268 (in 1.38 ms)
      Object {data: Object, status: 201...
    action LOG_IN_FULFILLED @ 01:53:31.390 (in 5.38 ms)

The Object in the middle is from the actual API response. It does work and my reducer receives and handles `LOG_IN_FULFILLED` as expected.

The issue is related to the presence of `redux-logger` middleware. When I have `redux-logger` before `redux-promise-middleware` when applying, like this:

    applyMiddleware([logger, thunk, promiseMiddleware()])

it leads to promiseMiddleware not sending the correct `_PENDING` action. When the logger is placed at the end of the middleware list, this doesn't happen. The owner of the `redux-logger` [rejected the issue](https://github.com/evgenyrodionov/redux-logger/issues/242).

**Comment by pburtchaell** [comment]
Thanks for the issue @nicnilov. This is expected behavior. 

Your`LOG_IN` action hits redux-logger before it hits redux-promise-middleware. When a pending action is dispatched by redux-promise-middleware, it doesn't go back through the chain of middleware. There's no way for it to ever hit redux-logger, hence it isn't logged.

Is there a reason you don't want to put redux-logger after redux-promise-middleware?

**Comment by nicnilov** [comment]
It makes sense that the logger being the first in middleware chain sees the action in its original form before the transformation to `_PENDING`. What bothered me is my reducer wasn't getting the `LOG_IN_PENDING` action. As it turned out it was an unrelated issue. It was a 12-hour run and I got confused. Re-checked today, and yes, the reducer gets the `LOGIN_PENDING` action even though the original `LOG_IN` gets logged, which is totally expected. My mistake, sorry to bother you.

I just placed the logger after redux-promise-middleware and it all looks good. Thanks!
